code/modules/xgm/xgm_gas_mixture.dm

```diff
@@ -19,10 +19,17 @@
 	///The last cached color of the gas mixture
 	var/tmp/cached_mix_color
 
-/datum/gas_mixture/New(_volume = CELL_VOLUME, _temperature = 0, _group_multiplier = 1)
-	volume = _volume
-	temperature = _temperature
-	group_multiplier = _group_multiplier
+/datum/gas_mixture/New(_volume, _temperature, _group_multiplier)
+	if(!isnull(_volume))
+		volume = _volume
+	if(!isnull(_temperature))
+		temperature = _temperature
+	if(!isnull(_group_multiplier))
+		group_multiplier = _group_multiplier
+
+	//Since we may have values defined on creation, update everything.
+	if(volume)
```

Comment:
Well, mainly because you need a structure around the datum that ensures update_value gets called on init, otherwise its data is invalid.
Like it doesn't guarantees it's own state is valid after init. But a proc that gets the datum doesn't know that. And it shouldn't have to know that. This is the kind of gotchas that creates obscure bugs that are hard to track down.

So, the problem we have is you can't just define a gas mix subtype with the gases and everything you want defined, you have to make a separate list to hold the gas values and feed those to the mixture at runtime. 

At the end of the day, if you skip updating values on new, it means you'll need to do several proc calls and update_values calls later on.

Reply:
So, I realized that we really didn't need to call update_values() when the gas list is empty. So I added a condition for that. It should basically cut all the extra overhead where it's not needed at all.